### ISSUE TITLE
starlark split/rsplit inconsistencies with spec

### ISSUE THREAD

**Issue by Quarz0** [bazelbuild/bazel#8726]
- It's mentioned in the [spec](https://github.com/bazelbuild/starlark/blob/master/spec.md#stringsplit) that `If sep is the empty string, split fails.`, however split is allowed to be called with `sep` as an empty string.
```python
"abc".split("")     #   ["a", "b", "c", ""]
```

- The spec also mentions `If sep is not specified or is None, split uses a different algorithm: ...`. However split does not have a default value for `sep` and complains if it is `None`.
```python
"abc".split()        ###  parameter 'sep' has no default value
```
```python
"abc".split(None)    ###  expected value of type 'string' for parameter 'sep'
```

**Comment by laurentlb** [comment]
For the empty separator, we're tracking it here: https://github.com/bazelbuild/bazel/issues/7355

I don't think it's useful to support split without argument. The behavior doesn't feel consistent.
cc @alandonovan

**Comment by brandjon** [comment]
The first case is fixed as per #7355.

Filed bazelbuild/starlark#176 to discuss the second case's design. Keeping this issue open to track the implementation.

**Comment by sgowroji** [comment]
Hi there! We're doing a clean up of old issues and will be closing this one. Please reopen if you’d like to discuss anything further. We’ll respond as soon as we have the bandwidth/resources to do so.
